src/components/battles/animations/shared/Fainting.ts

```diff
@@ -57,4 +67,31 @@ export class Fainting<TGameStartr extends FullScreenPokemon> extends Component<T
 
         this.gameStarter.modAttacher.fireEvent(this.gameStarter.mods.eventNames.onFaint, pokemon, battleInfo.teams.player.actors);
     }
-}
+
+    /**
+     * Sees if battle is against trainer or wild Pokemon.
+     *
+     * @param teamName    Tells if the Pokemon is the players or the opponents.
+     * @param battleInfo    Used to tell if player is fighting a wild Pokemon or trainer.
+     * @param pokemon   Information of defeated Pokemon.
+     */
+    private isThereATrainer(teamName: string, battleInfo: ITeamLeader | undefined , pokemon: IPokemon): IExperienceGains[] {
+        if (teamName === "opponent") {
+            let trainer: boolean = true;
```

Comment:
Make this `const` and directly set it to whether `battleInfo` is defined. Cleaner.